ext/opcache/zend_persist.c

```diff
@@ -76,6 +76,25 @@
 		} \
 	} while (0)
 
+#define zend_persist_attributes(attr) do { \
+	HashTable *ptr = zend_shared_alloc_get_xlat_entry(attr); \
+	if (ptr) { \
+		(attr) = ptr; \
+	} else { \
+		Bucket *p; \
+		zend_hash_persist(attr); \
+		ZEND_HASH_FOREACH_BUCKET((attr), p) { \
+			if (p->key) { \
+				zend_accel_store_interned_string(p->key); \
+			} \
+			zend_persist_zval(&p->val); \
+		} ZEND_HASH_FOREACH_END(); \
+		(attr) = zend_shared_memdup_put_free((attr), sizeof(HashTable)); \
+		GC_SET_REFCOUNT((attr), 2); \
+		GC_TYPE_INFO(attr) = IS_ARRAY | (IS_ARRAY_IMMUTABLE << GC_FLAGS_SHIFT); \
+	} \
+} while (0)
```

Comment:
Any reason to use a macro rather than a function here?

Reply:
It was a little more convenient than passing a HashTable pointer pointer I guess... I will refactor this into a function during implementation of attribute structs.